### PR TITLE
[C3TP-247] Prototype necessary workflow on jenkins deployment. 

### PR DESCRIPTION
**Pull request by AndreasAbdi**
This PR is to represent some of the work that is being done for C3TP. 

This specific PR contains necessary files to create a docker image that can be deployed to kubernetes. 

Added files and documentation on how to deploy.
Fixed error in numpy versioning. 
Added dockerfile to deploy as a docker container.

Added requirements.txt to install pre-reqs before running setup.py.

### CODE REVIEW

**Comment by andamian** [comment]
Thanks for updating the documentation and fixing the numpy version.

I don't think the requirements file is necessary as each of the two repos has its own development requirements file (dev_requirements.txt). Please update those and have docker instance call them.

**Comment by AndreasAbdi** [comment]
I'm seeing the dev_requirements.txt for the cadcdata repo but not for the cadcutils repo. 
Is it missing? 

I'll add one just in case.

**Comment by AndreasAbdi** [comment]
Updated the docker image to call the separate requirements.txt files from each repo. @andamian, if you could check if that's appropriate.

File: cadcutils/dev_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+numpy==1.12.0b1
+astropy==1.2.1
+cadcutils==1.0
```
Comment (andamian) [inline cadcutils/dev_requirements.txt]: cadcutils==1.0 not required as the following line installs the local version.

Please remove it from vofs/dev_requirements.txt too - thanks

File: README.md [hunk 2]
```diff
@@ -3,3 +3,76 @@ Client tools for using CADC services
...
+
+
+## Installing Packages
+
+Run 
+```
+cd cadcutils && pip install -r ./dev_requirements.txt
+cd cadcdata && pip install -r ./dev_requirements.txt
+pip install cadcutils/
+pip install cadcdata/
```
Comment (andamian) [inline README.md]: These two lines are redundant as these installs already occur in dev_requirements.txt

File: README.md [hunk 3]
```diff
@@ -3,3 +3,76 @@ Client tools for using CADC services
...
+```
+
+## Testing packages
+
+### Testing cadcutils
+
+```
+cd ./cadcutils
+pip install -r ./dev_requirements.txt
```
Comment (andamian) [inline README.md]: Redundant - see above

File: README.md [hunk 4]
```diff
@@ -3,3 +3,76 @@ Client tools for using CADC services
...
+python setup.py test
+```
+
+## Usage Example
+
+Run the following to install the packages 
+```
+pip install cadcutils/
+pip install cadcdata/
+```
```
Comment (andamian) [inline README.md]: They should be installed at this point already.

File: README.md [hunk 5]
```diff
@@ -3,3 +3,76 @@ Client tools for using CADC services
...
+
+```
+cd ./cadcdata
+pip install -r ./dev_requirements.txt
+pip install ../cadcutils/
+python setup.py test
+```
+
+## Usage Example
+
```
Comment (andamian) [inline README.md]: This is the usage example for the library mode. More examples are in the doc of the DataWsClient class.

To use cadc-data directly, one would do:
'cadc-data get GEMINI 00AUG02_002'

more info:
'cadc-data put --help'
'cadc-data get --help'
'cadc-data info --help'

File: cadcutils/dev_requirements.txt (hunk 1 above)
Comment (AndreasAbdi) [inline cadcutils/dev_requirements.txt]: Which one is vofs? 
I'm assuming this repo, so i'll send a pull request there. 
https://github.com/opencadc/vostools

File: README.md (hunk 5 above)
Comment (AndreasAbdi) [inline README.md]: Added the usage case for cadc-data as a command line utility. 

Unless you were asking I remove the usage case in library mode?

File: cadcutils/dev_requirements.txt (hunk 1 above)
Comment (AndreasAbdi) [inline cadcutils/dev_requirements.txt]: Didn't see it in the vofs repo, removed it from the cadcdata/dev_requirements.txt though.

**Comment by andamian** [comment]
Thanks for your contribution.

**Pull request merged**